### PR TITLE
Adding test for public interface

### PR DESCRIPTION
**Pull request by TotoGaz**
Adds tests that validate the public interface of `pvt` w.r.t. its previous results.
This validation is not very good but may help prevent from any silly mistake.

### CODE REVIEW

File: PVTPackage/source/tests/testPublicApi.cpp [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+TEST( pvt, publicApi )
+{
+  // FIXME Is there a simple way to use data providers with gtest?
+  const std::vector< std::string > fileNames{
+    "pvt_data.txt",
+    "4comp_2ph_1d.log.err",
+    "dead_oil_wells_2d.log.err",
+    "compositional_multiphase_wells_1d.log.err",
+    "compositional_multiphase_wells_2d.log.err",
+//    "deadoil_3ph_staircase_3d-10000.log.err", // TODO Waiting for more elements in the table.
```
Comment (francoishamon) [inline PVTPackage/source/tests/testPublicApi.cpp:248]: I will fix this when I merge my PR in the main repo.

File: PVTPackage/source/tests/testPublicApi.cpp [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+TEST( pvt, publicApi )
+{
+  // FIXME Is there a simple way to use data providers with gtest?
+  const std::vector< std::string > fileNames{
+    "pvt_data.txt",
+    "4comp_2ph_1d.log.err",
+    "dead_oil_wells_2d.log.err",
+    "compositional_multiphase_wells_1d.log.err",
+    "compositional_multiphase_wells_2d.log.err",
```
Comment (francoishamon) [inline PVTPackage/source/tests/testPublicApi.cpp:247]: I could not find these files in the PR. Do you grab them from the `integratedTests` repository or do you have a local copy?

**Review by francoishamon — APPROVED** [review]
Thanks very much @TotoGaz . To be honest I do not know much about serialization, deserialization, and json, but these tests look very useful!

I have some practical questions:
1) How can I run these tests on my machine? Is there a specific folder/a command to use?
2) If we find a bug in the `PVTPackage` that changes the results of the `testMultiFluid` test, how can we "rebaseline" and generate a new `pvt_data.txt` file in `source/tests/data`?
3) If we rebaseline the compositional integrated tests in the main repository, is there any impact on the `PVTPackage` tests?

File: PVTPackage/source/tests/testPublicApi.cpp (hunk 2 above)
Comment (TotoGaz) [reply PVTPackage/source/tests/testPublicApi.cpp]: I removed the files from the list. They were my files but they are too big to commit (there is some hammer/fly flavour) here.
I can send them to you if you want.

**Comment by TotoGaz** [comment]
> 1. How can I run these tests on my machine? Is there a specific folder/a command to use?

`testPublicApi` contains a unit test. Just run it (I did not manage to insert it with the other yet, and probably you should be careful about the `working directory`). This is not production grade yet.
Do not forget to define `ENABLE_PVTPACKAGE_TESTS`.
Parallel tests are just crap`^W` work in progress, forget about it for now. 😉 

> 2. If we find a bug in the `PVTPackage` that changes the results of the `testMultiFluid` test, how can we "rebaseline" and generate a new `pvt_data.txt` file in `source/tests/data`?

In `testPublicApi.cpp` there is this
```c++
  // Building new json reference, uncomment lines below to display computed values as json.
  auto newRef = json{
    { PublicAPIKeys::INPUT,  j.at( PublicAPIKeys::INPUT )},
    { PublicAPIKeys::OUTPUT, actualMsp }
  };
  std::cout << newRef << std::endl;
```
That you redisplay the new results.
Not automatic but we'll see how it proceeds.

> 3. If we rebaseline the compositional integrated tests in the main repository, is there any impact on the `PVTPackage` tests?

I'm not sure to fully understand but: `PVTPackage` now has a clear input output "protected" by those tests.
So as long as these input/output do not change, `PVTPackage` is not impacted (it does not have to know: `GEOSX` depends on `PVTPackage`, not the opposite)

**Pull request merged**